service/history/historyReplicator.go

```diff
@@ -77,32 +81,39 @@ func newHistoryReplicator(shard ShardContext, historyEngine *historyEngineImpl,
 }
 
 func (r *historyReplicator) ApplyEvents(request *h.ReplicateEventsRequest) (retError error) {
+	defer func() {
+		if _, ok := retError.(*shared.EntityNotExistsError); ok {
+			retError = errRetry
```

Comment:
This catch all error could make debugging little harder.  I think we should also include the actual message here.  Or atleast log a warning before returning the catch all error.